src/vmm/src/config.rs

```diff
@@ -1,13 +1,100 @@
 // Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
 
+use std::collections::HashMap;
 use std::convert::TryFrom;
 use std::fmt;
+use std::num;
 use std::path::PathBuf;
 use std::result;
 
 use super::{DEFAULT_HIGH_RAM_START, DEFAULT_KERNEL_CMDLINE};
 
+trait ArgParsing {
```

Comment:
Do you think  it's a good idea to use a name like `CfgArgParser` and `CfgArgParseError` ? They appear specific to config parsing, so slightly more 'explicit' sounds better.

Reply:
And then also may be rename the `fn` to something like `cfg_parse` as this is really doing that and `parse` is a standard library function  for [`str` ish types](https://doc.rust-lang.org/std/string/struct.String.html#method.parse) ?

Reply:
I am also thinking, if it makes sense to not have this all heavy lifting of defining a trait etc. A simple generic function should be able to take care of this. Something like the following - 

```rust 

fn cfg_param_parse<T: FromStr>(name: &str, value: String)  -> Result<T, <T as FromStr>::Err> {
    value.parse::<T>()
}
```
And then just `map_err` returned error if any to `ConversionError`. I believe all the values that will be parsed will be `FromStr` (Most of them will simply be strings, leaving behind some `int` and `PathBuf` types.) This should make the whole implementation considerably simple. 

in this [playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f84583eb6866afca0b4e73233972735a)

Reply:
> I am also thinking, if it makes sense to not have this all heavy lifting of defining a trait etc. A simple generic function should be able to take care of this. Something like the following -
> 
> ```rust
> fn cfg_param_parse<T: FromStr>(name: &str, value: String)  -> Result<T, <T as FromStr>::Err> {
>     value.parse::<T>()
> }
> ```
> 
> And then just `map_err` returned error if any to `ConversionError`. I believe all the values that will be parsed will be `FromStr` (Most of them will simply be strings, leaving behind some `int` and `PathBuf` types.) This should make the whole implementation considerably simple.
> 
> in this [playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f84583eb6866afca0b4e73233972735a)

This is actually a very good idea, thanks! Even if the current proposal should be more easily extensible for types that might need to do something more complex other then `parse`, I currently don't see any reason not to use your proposal instead. Thanks!